### ISSUE TITLE
Cannot use WebSockets if RequestFilter is set

### ISSUE THREAD

**Issue by unoexperto** [AsyncHttpClient/async-http-client#1523]
Hi @slandelle ! I hope you're around today :)

Turned out I can't use WebSockets if RequestFilter is set in AHC config. Here is what happens. There is following code in `NettyRequestSender`

https://github.com/AsyncHttpClient/async-http-client/blob/30e7235daccf4edcb2ce5639ebe4d62e9437a78a/client/src/main/java/org/asynchttpclient/netty/request/NettyRequestSender.java#L550-L564

it fails on `asyncHandler instanceof WebSocketUpgradeHandler` check because `asyncHandler` is instance of `com.sun.proxy.$Proxy0`. It happens because of this code in `DefaultAsyncHttpClient`

https://github.com/AsyncHttpClient/async-http-client/blob/5bbd787d7bc7be196e57a61cbd1f571e63b46bd8/client/src/main/java/org/asynchttpclient/DefaultAsyncHttpClient.java#L208-L220

which wraps my instance of `WebSocketUpgradeHandler` into proxy object.

What should I do ?

PS. I suspect same problem may happens here too https://github.com/AsyncHttpClient/async-http-client/blob/30e7235daccf4edcb2ce5639ebe4d62e9437a78a/client/src/main/java/org/asynchttpclient/netty/request/NettyRequestSender.java#L394

**Comment by slandelle** [comment]
Those Request/Response/IOException filters are an endless source of bugs (just crawl the bug tracker and I'll see what I mean).

They come from before I joined here and I'm really not fond of them (and I don't use them). IMO, they are can openers: usage is too wide and overlaps with what one could achieve with CompletageFuture or AsyncHandlers.

My plan is to kill them in the next major release. Try to find another way.

Closing as won't fix.
